helix-core/src/main/java/org/apache/helix/controller/stages/CustomizedViewAggregationStage.java

```diff
@@ -88,7 +93,9 @@ public void execute(final ClusterEvent event) throws Exception {
     // update customized view
     for (String stateType : customizedStateOutput.getAllStateTypes()) {
       List<CustomizedView> updatedCustomizedViews = new ArrayList<>();
+      Map<String, Map<Partition, Map<String, Long>>> updatedStartTimestamps = new HashMap<>();
       Map<String, CustomizedView> curCustomizedViews = new HashMap<>();
+      Map<String, CustomizedView> curCustomizedViewsCopy = new HashMap<>();
```

Comment:
As we discussed, since this stage is already async stage, we don't need to introduce these complicities, we can make the report logic synchronizely done in this stage.